Marlin/temperature.h

```diff
@@ -85,24 +95,64 @@ extern float current_temperature_bed;
 //inline so that there is no performance decrease.
 //deg=degreeCelsius
 
-FORCE_INLINE float degHotend(uint8_t extruder) { return current_temperature[extruder]; }
-FORCE_INLINE float degBed() { return current_temperature_bed; }
+FORCE_INLINE float degHotend(uint8_t extruder) {  
+#ifndef SINGLENOZZLE
+  return current_temperature[extruder];
+#else
+  return current_temperature[0];
+#endif
+}
```

Comment:
Here you can be more concise with:

    #ifdef SINGLENOZZLE
      #define EXTRUDER_ARG 0
    #else
      #define EXTRUDER_ARG extruder
    #endif

    FORCE_INLINE float degHotend(uint8_t extruder) { return current_temperature[EXTRUDER_ARG]; }
    FORCE_INLINE float rawHotendTemp(uint8_t extruder) { return current_temperature_raw[EXTRUDER_ARG]; }
    ...

...etc...